File: python/7.整数反转.py
#
# @lc app=leetcode.cn id=7 lang=python3
#
# [7] 整数反转
#

# @lc code=start
class Solution:
    def reverse(self, x: int) -> int:
        s = (x > 0) - (x < 0) 
        r = int(str(abs(x))[::-1])
        return s*r (r<2**31)
        
# @lc code=end

File: C++/7.整数反转.cpp
/*
 * @lc app=leetcode.cn id=7 lang=cpp
 *
 * [7] 整数反转
 */

// @lc code=start
class Solution {
public:
    int reverse(int x) {
        int flag = x;
        int res = 0;
        if (x == INT_MIN) {
            return 0;
        }
        if (x < 0) {
            x = -x;
        }
        while ( x > 0) {
            int tmp = x % 10;
            x = x / 10;
            if (res > INT_MAX / 10 || (res == INT_MAX / 10 && tmp > 7)) {
                return 0;
            }
            res = res * 10 + tmp;
        }
        return flag > 0 ? res : -res;
    }
};
// @lc code=end

